src/test/java/bookbob/parser/ParserTest.java

```diff
@@ -0,0 +1,978 @@
+        void testFindVisitCommand() throws IOException {
+            // Arrange
+            String input = "findVisit S9534567A";
+
+            // Act
+            boolean result = Parser.handleCommand(input, commandHandler, records, appointmentRecord);
+
+            // Assert
+            Mockito.verify(commandHandler, Mockito.times(1)).findVisitByIc("S9534567A", records);
+            Assertions.assertTrue(result);
+        }
+    }
+
+    //@@author kaboomzxc
+    @Nested
+    @DisplayName("Appointment Command Tests")
+    class AppointmentCommandTests {
+        @Test
+        @DisplayName("Test add appointment command")
+        void testAddAppointmentCommand() throws IOException {
+            // Arrange
+            String input = "appointment n/John Doe ic/S9534567A date/21-10-2024 time/15:48";
+
+            // Act
+            boolean result = Parser.handleCommand(input, commandHandler, records, appointmentRecord);
+
+            // Assert
+            Mockito.verify(commandHandler, Mockito.times(1)).appointment(input, appointmentRecord);
+            Assertions.assertTrue(result);
+        }
+
+        //@@author kaboomzxc
+        @Test
+        @DisplayName("Test list appointments command")
+        void testListAppointmentsCommand() throws IOException {
+            // Arrange
+            String input = "listAppointments";
+
+            // Act
+            boolean result = Parser.handleCommand(input, commandHandler, records, appointmentRecord);
+
+            // Assert
+            Mockito.verify(commandHandler, Mockito.times(1)).listAppointments(appointmentRecord);
+            Assertions.assertTrue(result);
+        }
+
+        //@@author kaboomzxc
+        @Test
+        @DisplayName("Test find appointment command")
+        void testFindAppointmentCommand() throws IOException {
+            // Arrange
+            String input = "findAppointment ic/S9534567A";
+
+            // Act
+            boolean result = Parser.handleCommand(input, commandHandler, records, appointmentRecord);
+
+            // Assert
+            Mockito.verify(commandHandler, Mockito.times(1)).findAppointment("ic/S9534567A", appointmentRecord);
+            Assertions.assertTrue(result);
+        }
+
+        //@@author kaboomzxc
+        @Test
+        @DisplayName("Test delete appointment command")
+        void testDeleteAppointmentCommand() throws IOException {
+            // Arrange
+            String input = "deleteAppointment ic/S9534567A date/21-10-2024 time/15:48";
+
+            // Act
+            boolean result = Parser.handleCommand(input, commandHandler, records, appointmentRecord);
+
+            // Assert
+            Mockito.verify(commandHandler, Mockito.times(1)).deleteAppointment(input, appointmentRecord);
+            Assertions.assertTrue(result);
+        }
+    }
+
+    //@@author kaboomzxc
+    @Nested
+    @DisplayName("Search Command Tests")
+    class SearchCommandTests {
```

Comment:
Perhaps can rename this as FindVisitCommandTests

Reply:
actually for this part, not really testing findVisit only, was also testing "findDiagnosis" and "findMedication" also

but i renamed to "find" which suits better

Reply:
Yup sure